libs/driver/src/VideoDriver.cpp

```diff
@@ -72,5 +75,54 @@ VideoMode VideoDriver::FindClosestVideoMode(const VideoMode& mode) const
 void VideoDriver::SetNewSize(VideoMode windowSize, Extent renderSize)
 {
     windowSize_ = windowSize;
-    renderSize_ = renderSize;
+    screenRenderSize_ = renderSize;
+    viewRenderSize_ = guiScale_.screenToView<Extent>(renderSize);
+
+    const auto ratioXY = PointF(screenRenderSize_) / PointF(windowSize_.width, windowSize_.height);
+    dpiScale_ = (ratioXY.x + ratioXY.y) * .5f; // use the average ratio of both axes
+}
+
+void VideoDriver::setGuiScale(unsigned percent)
+{
+    if(percent == 0)
+        getGuiScaleRange(nullptr, nullptr, &percent);
+
+    if(guiScale_.percent() == percent)
+        return;
+
+    // translate current mouse position to screen space
+    const auto screenPos = guiScale_.viewToScreen(mouse_xy.pos);
+
+    guiScale_ = GuiScale(percent);
+
+    onGuiScaleChanged(); // give backends an opportunity to update their state
+
+    CallBack->WindowResized();
+
+    // move cursor to new position in view space
+    // must happen after window resize event to avoid drawing spurious hover events
+    mouse_xy.pos = guiScale_.screenToView(screenPos);
+    CallBack->Msg_MouseMove(mouse_xy);
+}
+
+void VideoDriver::getGuiScaleRange(unsigned* minPercent, unsigned* maxPercent, unsigned* recommendedPercent) const
+{
+    if(minPercent)
+        *minPercent = 100u;
```

Comment:
This is partly a historic artifact and partly to have one authoritative location to set the minimum. I originally had 50% DPI-adjusted here and then decided after feedback from testing, that 50% is definitely too low. Someday we may still decide to support values lower than 100% (maybe 80%?) and this would be the place to change it.

Reply:
Ok, agreed to keep it for future use